### PR TITLE
Make the state of UI and user metadata persistent

### LINKED ISSUE #44: Change return type of arguments in checkAndDisplayValidUser()
Change the return type to the expected return type of the set... functions

### PR DESCRIPTION
**Pull request by kpattaswamy**
This PR lets the state of the UI and the user's metadata persist in storage after closing (not logging out) the extension. 

The way it currently works:
Any time a set... function is called, a write will be made to chrome's local storage with a key/value pair representing the data being set. 

Upon creation of a new app object, the constructor will attempt to get data for the state of the UI and user metadata. If it is successful, it will reproduce the UI and re-instantiate the s3 object. Otherwise, the user will have to start by logging in again.

Keys are tied to chrome's session storage, they will be terminated as soon as the window is closed if the user doesn't logout. This form of security seems to be basic and it might even be worth encrypting keys and then storing it in the session storage. 

Resolves #44 , #39

### CODE REVIEW

File: src/app.tsx [hunk 1]
```diff
@@ -12,47 +14,104 @@ type Props = {
...
 
     // Function that will be passed as prop to globalize the S3 Object being made
-    setS3Obj(s3Obj:MyS3Auth){
-        this.setState({s3Obj})
+    setS3Obj(s3Obj:MyS3Auth | null){
+
+        this.setState({s3Obj});
+
+        // Change the user metadata only if the s3 object is not null
+        if (s3Obj){
```
Comment (KevinMi2023p) [inline src/app.tsx:51]: Should we say !== null for clarity?

**Review by KevinMi2023p — APPROVED** [review]
Looks good to me

File: src/app.tsx (hunk 1 above)
Comment (kpattaswamy) [reply src/app.tsx:51]: Actually that means something different 

see this https://stackoverflow.com/questions/4361585/how-to-check-if-a-variable-is-not-null

Comment (kpattaswamy) [reply src/app.tsx:51]: imo I think its more concise

**Pull request merged**